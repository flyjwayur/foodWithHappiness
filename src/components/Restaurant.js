import React from "react";
import { Blurhash } from "react-blurhash";
import data from "../data/restaurants.json";

const Restaurant = () => (
    <div>
      {data.restaurants.map((restaurant, i) => {
        return (
          <div key={i}>
            <div>{restaurant.name}</div>
            <div>{restaurant.description}</div>
            <div>{restaurant.delivery_price}</div>
            <Blurhash
              hash={restaurant.blurhash}
              width={400}
              height={300}
              resolutionX={32}
              resolutionY={32}
              punch={1}
            />
            <img src={restaurant.image} alt={restaurant.name + "_image"} />
          </div>
        );
      })}
    </div>
  );

export default Restaurant;
