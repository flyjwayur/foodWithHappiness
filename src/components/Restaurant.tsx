import React, { useState, FunctionComponent } from 'react';
import { Blurhash } from 'react-blurhash';
import ContentsLoader from './ContentsLoader';
import { IPropRestaurant } from '../index.d';
import styled from 'styled-components';
import { mediaQuerySizes } from '../theme/mediaQuery';
import theme from '../theme/theme';

const Card = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 20rem;
  border-radius: 8px;
  box-shadow: 0px 0px 10px 0px #aaa;

  > img {
    width: 100%;
    height: 12rem;
    object-fit: cover;
  }

  @media (max-width: ${mediaQuerySizes.tabletPortrait}) {
    min-width: 20rem;
  }
`;

const RestaurantImage = styled.img`
  border-radius: 8px 8px 0 0;
`;

const CardTextTop = styled.div`
  flex: 1 1 auto;
  min-width: 0;
  padding: 0.875rem 1rem;
  padding-bottom: 1rem;
  background-color: ${theme.lightGreyBackground};
`;

const CardTextTitle = styled.div`
  font-weight: 500;
  font-size: 1.125rem;
  color: ${theme.greyText};
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;

const CardTextDesc = styled.div`
  font-weight: 400;
  font-size: 1rem;
  color: #868780;
  margin-top: 2px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;

const CardTextBottom = styled.div`
  border-top: 1px dashed #e4e4e4;
  font-size: 12px;
  color: #868789;
  padding: 0.5rem 1rem;
  display: flex;
  justify-content: space-between;
`;

const DeliveryPrice = styled.span`
  font-size: 0.8rem;
  line-height: 1.3rem;
`;

const Tags = styled.div``;

const Tag = styled.span`
  width: max-content;
  background-color: ${theme.primary};
  color: ${theme.lightText};
  text-align: center;
  font-size: 0.8rem;
  line-height: 1.3rem;
  border-radius: 8px;
  padding: 0.3rem 0.5rem;
  margin: 0.3rem;
`;

const Restaurant: FunctionComponent<{
  restaurant: IPropRestaurant;
  isContentsLoading: boolean;
}> = ({ restaurant, isContentsLoading }) => {
  const {
    name,
    description,
    delivery_price,
    blurhash,
    image,
    tags,
    online,
    currency,
    city
  } = restaurant;
  const [isImageLoading, setIsImageLoading] = useState(true);

  return (
    <>
      {isContentsLoading ? (
        <ContentsLoader />
      ) : (
        <Card>
          <Blurhash
            style={{ display: isImageLoading ? 'inline-block' : 'none' }}
            hash={blurhash}
            width={400}
            height={300}
            resolutionX={32}
            resolutionY={32}
            punch={1}
            onLoad={() => console.log('blurhash')}
          />
          <RestaurantImage
            style={{ display: isImageLoading ? 'none' : 'inline-block' }}
            src={image}
            alt={name + '_image'}
            width="400px"
            height="300px"
            onLoad={() => setIsImageLoading(false)}
          />
          <CardTextTop>
            <CardTextTitle>{name}</CardTextTitle>
            <CardTextDesc>{description}</CardTextDesc>
          </CardTextTop>
          <CardTextBottom>
            <DeliveryPrice>{delivery_price} â‚¬</DeliveryPrice>
            <Tags>
              {tags.map((tag, i) => (
                <Tag key={tag + i}>{tag}</Tag>
              ))}
            </Tags>
            {/* <span>{online}</span>
            <span>{city}</span> */}
          </CardTextBottom>
        </Card>
      )}
    </>
  );
};

export default Restaurant;
