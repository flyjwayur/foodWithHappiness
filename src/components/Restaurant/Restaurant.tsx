import React, { useState, FunctionComponent } from 'react';
import Blurhash from '../Blurhash/Blurhash';
import ContentsLoader from '../ContentsLoader/ContentsLoader';
import { IPropRestaurant } from '../../index.d';
import styled from 'styled-components';
import { mediaQuerySizes } from '../../theme/mediaQuery';
import theme from '../../theme/theme';
import { formatCurrency } from '../../util/formatCurrency';

const Card = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 20rem;
  border-radius: 8px;
  box-shadow: 0px 0px 10px 0px #aaa;

  > img {
    width: 100%;
    height: 12rem;
    object-fit: cover;
  }

  @media (max-width: ${mediaQuerySizes.tabletLandscape}) {
    min-width: 25rem;
  }

  @media (max-width: ${mediaQuerySizes.tabletPortrait}) {
    min-width: 19rem;
  }
`;

const RestaurantImage = styled.img`
  border-radius: 8px 8px 0 0;
  width: 100%;
`;

const CardTextTop = styled.div`
  flex: 1 1 auto;
  min-width: 0;
  padding: 0.875rem 1rem;
  padding-bottom: 1rem;
  background-color: ${theme.lightGreyBackground};
`;

const CardTextTitle = styled.div`
  font-weight: 500;
  font-size: 1.125rem;
  color: ${theme.darkgreyText};
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;

const CardTextDesc = styled.div`
  font-weight: 400;
  font-size: 1rem;
  color: #868780;
  margin-top: 2px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
`;

const CardTextBottom = styled.div`
  border-top: 1px dashed ${theme.linePrimaryColor};
  font-size: 12px;
  color: #868789;
  padding: 0.5rem 1rem;
  display: flex;
  justify-content: space-between;
`;

const DeliveryPrice = styled.span`
  font-size: 0.8rem;
  line-height: 1.3rem;
`;

const Tags = styled.div``;

const Tag = styled.span`
  width: max-content;
  background-color: ${theme.primary};
  color: ${theme.lightText};
  text-align: center;
  font-size: 0.8rem;
  line-height: 1.3rem;
  border-radius: 5px;
  padding: 0.3rem 0.5rem;
  margin: 0.3rem;
`;

const Restaurant: FunctionComponent<{
  restaurant: IPropRestaurant;
  isContentsLoading: boolean;
}> = ({ restaurant, isContentsLoading }) => {
  const { name, description, delivery_price, blurhash, image, tags } = restaurant;
  const [isImageLoading, setIsImageLoading] = useState(true);

  return (
    <>
      {isContentsLoading ? (
        <ContentsLoader />
      ) : (
        <Card data-testid="restaurant">
          <Blurhash
            style={{ display: isImageLoading ? 'inline-block' : 'none' }}
            hash={blurhash}
            width="100%"
            height="100%"
            resolutionX={32}
            resolutionY={32}
            punch={1}
          />
          <RestaurantImage
            style={{ display: isImageLoading ? 'none' : 'inline-block' }}
            src={image}
            alt={name + '_image'}
            onLoad={() => setIsImageLoading(false)}
          />
          <CardTextTop>
            <CardTextTitle>{name}</CardTextTitle>
            <CardTextDesc>{description}</CardTextDesc>
          </CardTextTop>
          <CardTextBottom>
            <DeliveryPrice> Delivery {formatCurrency(delivery_price)} â‚¬</DeliveryPrice>
            <Tags>
              {tags.map((tag, i) => (
                <Tag key={tag + i}>{tag}</Tag>
              ))}
            </Tags>
          </CardTextBottom>
        </Card>
      )}
    </>
  );
};

export default Restaurant;
