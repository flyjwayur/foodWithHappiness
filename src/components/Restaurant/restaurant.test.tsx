import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Restaurant from './Restaurant';
import 'jest-styled-components';

afterEach(cleanup);

describe('<Restaurant />', () => {
  const dummyRestaurtant = {
    blurhash: 'UUKJMXv|x]oz0gtRM{V@AHRQwvxZXSs9s;o0',
    city: 'Helsinki',
    currency: 'EUR',
    delivery_price: 390,
    description: 'Asenneburgeri',
    image:
      'https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90',
    location: [24.941325187683105, 60.169938852212965],
    name: 'Social Burgerjoint Citycenter',
    online: false,
    tags: ['hamburger', 'fries']
  };

  test('Display restaurant info', async () => {
    // Arrange
    const isContentsLoading = false;

    const { findByTestId } = render(
      <Restaurant restaurant={dummyRestaurtant} isContentsLoading={isContentsLoading} />
    );

    const restaurant = await findByTestId('restaurant');

    // Act
    // Assert
    expect(restaurant).toMatchSnapshot();
  });
});
