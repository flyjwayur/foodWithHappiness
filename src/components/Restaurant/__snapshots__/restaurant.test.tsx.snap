// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Restaurant /> Display restaurant info 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 20rem;
  border-radius: 8px;
  box-shadow: 0px 0px 10px 0px #aaa;
}

.c0 > img {
  width: 100%;
  height: 12rem;
  object-fit: cover;
}

.c1 {
  border-radius: 8px 8px 0 0;
  width: 100%;
}

.c2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  padding: 0.875rem 1rem;
  padding-bottom: 1rem;
  background-color: #eff1f3;
}

.c3 {
  font-weight: 500;
  font-size: 1.125rem;
  color: #3c3d40;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c4 {
  font-weight: 400;
  font-size: 1rem;
  color: #868780;
  margin-top: 2px;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c5 {
  border-top: 1px dashed #e4e4e4;
  font-size: 12px;
  color: #868789;
  padding: 0.5rem 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-size: 0.8rem;
  line-height: 1.3rem;
}

.c7 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  background-color: #009de0;
  color: #ffffff;
  text-align: center;
  font-size: 0.8rem;
  line-height: 1.3rem;
  border-radius: 5px;
  padding: 0.3rem 0.5rem;
  margin: 0.3rem;
}

@media (max-width:1200px) {
  .c0 {
    min-width: 25rem;
  }
}

@media (max-width:900px) {
  .c0 {
    min-width: 19rem;
  }
}

<div
  class="c0"
  data-testid="restaurant"
>
  <div
    style="display: inline-block; height: 100%; width: 100%; position: relative;"
  >
    <canvas
      height="32"
      punch="1"
      style="position: absolute; top: 0px; bottom: 0px; left: 0px; right: 0px; width: 100%; height: 100%;"
      width="32"
    />
  </div>
  <img
    alt="Social Burgerjoint Citycenter_image"
    class="c1"
    src="https://prod-wolt-venue-images-cdn.wolt.com/5b348b31fe8992000bbec771/2be8c7738b220df2f9a0974da5c90d90"
    style="display: none;"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      Social Burgerjoint Citycenter
    </div>
    <div
      class="c4"
    >
      Asenneburgeri
    </div>
  </div>
  <div
    class="c5"
  >
    <span
      class="c6"
    >
       Delivery 
      3.90
       â‚¬
    </span>
    <div
      class=""
    >
      <span
        class="c7"
      >
        hamburger
      </span>
      <span
        class="c7"
      >
        fries
      </span>
    </div>
  </div>
</div>
`;
